ui/src/services/k8s/K8SLocalVolumeProvider.test.ts

```diff
@@ -0,0 +1,130 @@
+import { CoreV1Api, CustomObjectsApi } from '@kubernetes/client-node';
+import K8SLocalVolumeProvider from './K8SLocalVolumeProvider';
+import { Metalk8sV1alpha1VolumeClient } from './Metalk8sVolumeClient.generated';
+import { updateApiServerConfig } from '../k8s/api';
+
+jest.mock('../k8s/api', () => ({
+  updateApiServerConfig: jest.fn(),
+}));
+
+describe('K8SLocalVolumeProvider', () => {
+  let provider: K8SLocalVolumeProvider;
+  const mockUrl = 'mock-url';
+  const mockToken = 'mock-token';
+
+  const mockCustomObjectsApi = {
+    listClusterCustomObject: jest.fn(),
+  } as unknown as CustomObjectsApi;
+
+  const mockVolumeClient = new Metalk8sV1alpha1VolumeClient(
+    mockCustomObjectsApi,
+  );
+
+  const mockCoreV1Api = {
+    listNode: jest.fn(),
+    listPersistentVolume: jest.fn(),
+  } as unknown as CoreV1Api;
+
+  beforeEach(() => {
+    (updateApiServerConfig as jest.Mock).mockReturnValue({
+      coreV1: mockCoreV1Api,
+      customObjects: mockCustomObjectsApi,
+    });
+
+    provider = new K8SLocalVolumeProvider(mockUrl, mockToken);
+    provider.k8sClient = mockCoreV1Api;
+    provider.volumeClient = mockVolumeClient;
+  });
+
+  describe('listLocalPersistentVolumes', () => {
+    it('should return local persistent volumes for a given node.', async () => {
+      (mockCoreV1Api.listNode as jest.Mock).mockResolvedValue({
+        body: {
+          items: [
+            {
+              metadata: { name: 'test-node' },
+              status: {
+                addresses: [
+                  { type: 'Hostname', address: 'test-node' },
+                  { type: 'InternalIP', address: '192.168.1.100' },
+                ],
+              },
+            },
+          ],
+        },
+      });
+
+      (mockCoreV1Api.listPersistentVolume as jest.Mock).mockResolvedValue({
+        body: {
+          items: [
+            {
+              metadata: { name: 'test-volume' },
+              spec: {},
+            },
+          ],
+        },
+      });
+
+      (
+        mockCustomObjectsApi.listClusterCustomObject as jest.Mock
+      ).mockResolvedValue({
+        body: {
+          items: [
+            {
+              metadata: { name: 'test-volume' },
+              spec: {
+                nodeName: 'test-node',
+                rawBlockDevice: { devicePath: '/dev/sda' },
+              },
+            },
+          ],
+        },
+      });
+
+      const volumes = await provider.listLocalPersistentVolumes('test-node');
+
+      expect(volumes).toHaveLength(1);
+      expect(volumes[0]).toMatchObject({
+        IP: '192.168.1.100',
+        devicePath: '/dev/sda',
+        metadata: { name: 'test-volume' },
```

Comment:
No need for the metadata layer, the name could be at the root and changed to pvName